spec/src/main/doc/events.asciidoc

```diff
@@ -122,7 +148,9 @@ If two instances of the same qualifier type are passed to +select()+, an +Illega
 
 If an instance of an annotation that is not a qualifier type is passed to +select()+, an +IllegalArgumentException+ is thrown.
 
-The method +fire()+ fires an event with the specified qualifiers and notifies observers, as defined by <<observer_notification>>. If the container is unable to resolve the parameterized type of the event object, it uses the specified type to infer the parameterized type of the event types.
+The methods +fire()+ and `fireAsync()` fire an event with the specified qualifiers and notifies observers, as defined by <<observer_notification>>. If the container is unable to resolve the parameterized type of the event object, it uses the specified type to infer the parameterized type of the event types.
+
+The method `fireAsync()` may be called with a specific `Executor` object to be used to invoke observers method. The container should provide a default `Executor` when `fireAsync()` is called without specifying one.
```

Comment:
We should specify a case when the given executor is not capable of executing observer notifications (e.g. shut down). Should the fireAsync method throw something right away?